### PR TITLE
Update psycopg2 in requirements.txt

### LINKED ISSUE #75: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use "pip install psycopg2-binary" instead.
I currently get an email for every cronjob run because the cronjobs generate output on stderr. The output is:

```
/usr/local/lib/python3.6/dist-packages/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use "pip install psycopg2-binary" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.
  """)
```

From the linked documentation, it seems we should change the entry for `psycopg2` entry in requirements.txt to `psycopg2>=2.7,<2.8 --no-binary psycopg2`.

### PR DESCRIPTION
**Pull request by arjunsalyan**
Closes: https://github.com/macports/macports-webapp/issues/75

### CODE REVIEW

File: app/requirements.txt [hunk 1]
```diff
@@ -25,7 +25,7 @@ lazy-object-proxy==1.4.1
 MarkupSafe==1.1.1
 mccabe==0.6.1
 pluggy==0.12.0
-psycopg2==2.7.7
+psycopg2>=2.7,<2.8 --no-binary psycopg2
```
Comment (neverpanic) [inline app/requirements.txt]: I was under the impression that we wanted to specifically fix the versions to ensure our built containers are reproducible, so maybe we should make this `psycopg2==2.7.7 --no-binary psycopg2` instead (assuming that would work). Also, should we update to 2.8?

**Comment by umeshksingla** [comment]
I'm not sure if I can provide any feedback on this. When we started 2.7.7 was the latest (stable) version available with beta versions for 2.8. The [release notes for 2.8](http://initd.org/psycopg/articles/2019/04/04/psycopg-28-released/) do not report any major changes and we can benefit (in the near future) from the enhancement it made for throwing python-compatible exceptions.
But let's definitely fix its version in the requirements as Clemens said.

**Comment by arjunsalyan** [comment]
I have tested most of the functionalities for `django==2.2.4` and `psycopg2==2.8.3`. And it seems that we are good to upgrade these two their latest releases.

**Comment by neverpanic** [comment]
LG2M now, but shouldn't you squash those commits since they're somewhat related?

**Comment by arjunsalyan** [comment]
Oh, I just noticed that building of the image now fails on docker hub.
```
Collecting psycopg2==2.8.3 (from -r /code/app/requirements.txt (line 28))
Downloading https://files.pythonhosted.org/packages/5c/1c/6997288da181277a0c29bc39a5f9143ff20b8c99f2a7d059cfb55163e165/psycopg2-2.8.3.tar.gz (377kB)
ERROR: Command errored out with exit status 1:
command: /usr/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-5u6h49o7/psycopg2/setup.py'"'"'; __file__='"'"'/tmp/pip-install-5u6h49o7/psycopg2/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base pip-egg-info
cwd: /tmp/pip-install-5u6h49o7/psycopg2/
Complete output (23 lines):
running egg_info
creating pip-egg-info/psycopg2.egg-info
writing pip-egg-info/psycopg2.egg-info/PKG-INFO
writing dependency_links to pip-egg-info/psycopg2.egg-info/dependency_links.txt
writing top-level names to pip-egg-info/psycopg2.egg-info/top_level.txt
writing manifest file 'pip-egg-info/psycopg2.egg-info/SOURCES.txt'
Error: pg_config executable not found.
pg_config is required to build psycopg2 from source. Please add the directory
containing pg_config to the $PATH or specify the full executable path with the
option:
python setup.py build_ext --pg-config /path/to/pg_config build ...
or with the pg_config option in 'setup.cfg'.
If you prefer to avoid building psycopg2 from source, please install the PyPI
'psycopg2-binary' package instead.
For further information please check the 'doc/src/install.rst' file (also at
<http://initd.org/psycopg/docs/install.html>).
```

**Comment by arjunsalyan** [comment]
It has been fixed, `psycopg2` required `python-dev` and `libpq-dev` for successful compilation. The [development website](http://ec2-52-34-234-111.us-west-2.compute.amazonaws.com) is now running on these configurations, everything works fine.

**Review by mojca — APPROVED** [review]
Feel free to merge yourself. (I'm on my phone)

**Pull request merged**